### ISSUE TITLE
Condition searching throughout

### ISSUE THREAD

**Issue by brynrhodes** [cqframework/opioid-cds-r4#84]
US Core Searching Capability for Condition documents searching by Code as a SHOULD, but some implementations do not support that search capability. As such, we should fall back to the Category code for searching.

In addition to the categories defined in the base specification (problem-list, encounter-diagnosis), US Core defines health-concern as a potential condition category. It's not clear how these categories partition the patient's Condition resources, so we should always search for all categories.

As a result, all Condition references in the logic should be updated to:

```
valueset "US Core Condition Category Codes": 'http://hl7.org/fhir/us/core/ValueSet/us-core-condition-category'

// Condition searches like this:
[Condition: code in "Value Set X"] C ...
// become
[Condition: category in "US Core Condition Category Codes"] C
  where C.code in "Value Set X"
```

**Comment by sliver007** [comment]
Currently unable to filter on category as part of the retrieve so it's being done in the cql. This is due to the CQL Engine bug captured here: https://github.com/DBCG/cql_engine/issues/388

**Comment by sliver007** [comment]
Epic only allows for a single category when searching so we changed approaches to do a retrieve for each of the three categories specified in the US Core Condition Category value set then uniting them client-side.

**Comment by sliver007** [comment]
Resolved in GH# 6489a0e65685801704786628a559ebfb1168fb51
